### ISSUE TITLE
namespaced cookie error when visiting site

### ISSUE THREAD

**Issue by jonnymbgood** [cypress-io/cypress#2618]
### Current behavior:
A integration test with the single line `cy.visit('https://www.ancestry.com/')` causes the error `Error: Sorry, you cannot set a Cypress namespaced cookie.`

![screen shot 2018-10-16 at 10 10 52 am](https://user-images.githubusercontent.com/485900/47030830-ca24ac00-d12b-11e8-8689-7979b5dbe3d9.png)

Stack Trace:
```text
Command:   visit
Error:     CypressError: cy.visit() failed trying to load:

https://www.ancestry.com/

We attempted to make an http request to this URL but the request failed without a response.

We received this error at the network level:

  > Error: Sorry, you cannot set a Cypress namespaced cookie.

Common situations why this would fail:
  - you don't have internet access
  - you forgot to run / boot your web server
  - your web server isn't accessible
  - you have weird network configuration settings on your computer

The stack trace for this error is:

Error: Sorry, you cannot set a Cypress namespaced cookie.
    at Object.setCookie (/Users/jclark/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/automation/cookies.js:73:17)
    at /Users/jclark/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/automation/index.js:98:30
    at tryCatcher (/Users/jclark/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/util.js:16:23)
    at Function.Promise.attempt.Promise.try (/Users/jclark/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/method.js:39:29)
    at normalize (/Users/jclark/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/automation/index.js:89:30)
    at /Users/jclark/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/automation/index.js:142:20
    at tryCatcher (/Users/jclark/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Users/jclark/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:510:31)
    at Promise._settlePromise (/Users/jclark/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:567:18)
    at Promise._settlePromiseCtx (/Users/jclark/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:604:10)
    at Async._drainQueue (/Users/jclark/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/async.js:138:12)
    at Async._drainQueues (/Users/jclark/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/Users/jclark/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:781:20)
    at tryOnImmediate (timers.js:743:5)
    at processImmediate [as _immediateCallback] (timers.js:714:5)
```

### Desired behavior:
It should not throw an error

### Steps to reproduce:
Clone this [reproduction repo](https://github.com/jonnymbgood/cypress-issue-2618) and follow the README

or

Run this integration test:
```javascript
describe('Failure', function() {
    it("should work", function() {
        cy.visit('https://www.ancestry.com/')
    })
})
```
### Versions
Cypress 3.1.0
Chrome 69
Mac OS 10.13.6

**Comment by jennifer-shehane** [comment]
May be related to this cookie issue: https://github.com/cypress-io/cypress/issues/1321

**Comment by jonnymbgood** [comment]
@jennifer-shehane It does look close, except the resulting behavior is a bit different. I think the cause may be a bit different as well, but I'm not sure.

The culprit value in the `jar` variable in `setJarCookies()` is `Cookie="; Path=/; hostOnly=true; aAge=6ms; cAge=6ms`, so it looks like both the name AND value are missing (A "normal" cookie looks like `Cookie="___utmvmzcuXLwFB=pOXZxgqCRYs; Max-Age=900; Path=/; hostOnly=true; aAge=3ms; cAge=3ms`.

**Comment by jonnymbgood** [comment]
@jennifer-shehane I have verified that adding 
```javascript
if (!cookie.name) {
  return;
}
```
to `setJarCookies()` works. It would probably be better to ensure this never gets into the actual cookie jar, but I'm not sure it's worth opening a request in [tough-cookie](https://github.com/salesforce/tough-cookie) to filter these out.

Would you like me to open a PR with this fix?

**Comment by jonnymbgood** [comment]
@jennifer-shehane 
I have created a reproduction repo [here](https://github.com/jonnymbgood/cypress-issue-2618). All it takes to reproduce is add the header `Set-Cookie` with a blank value.

**Comment by jennifer-shehane** [comment]
@jonnymbgood Yes, opening a PR with a proposed fix is the fastest way that you will get your issue resolved in Cypress.

**Comment by jennifer-shehane** [comment]
Duplicate of https://github.com/cypress-io/cypress/issues/2659

This was fixed in [3.1.1](https://docs.cypress.io/guides/references/changelog.html#3-1-1)
